## Exploratory Data Analysis of Avalanche Observations pkl

Sarah Braverman

sz = start zone
<br />use AAA SWAG pdf on drive

In [2]:
import pandas as pd

In [22]:
df = pd.read_pickle("./data-prep/pkl/avy_obs.pkl")
df

avrid  avpathid  avobsdate avobstime avobserverid avtype  avtriggertyp  \
0        1    SHED10 2012-03-26  12:45:00     stein_01     WS           5.0   
1        2  PATH1163 2012-03-16  08:30:00     stein_01      U           5.0   
2        3     CURLY 2012-03-16  08:34:00     stein_01      U           5.0   
3        4    SHED10 2012-03-10  14:50:00     stein_01     WL           5.0   
4        5  INFINITY 2012-03-10  14:54:00     stein_01      L           5.0   
..     ...       ...        ...       ...          ...    ...           ...   
486    504   SHED07W 2022-03-01  05:04:00     clark_01     SS           5.0   
487    505   AP1163H 2022-03-23  13:30:00     clark_01     WL           5.0   
488    506    SHED10 2022-03-23  13:30:00     clark_01     WL           5.0   
489    507   MCAMSER 2022-03-23  18:00:00     clark_01     HS           5.0   
490    508  PATH1163 2022-03-27  14:35:00     clark_01     WL           5.0   

    avtrigger avrelativesize avdestructivesize  ... deb_hwy deb_hwy_w  \
0          NL             R1              D1.5  ...   False       0.0   
1          NU              U                D2  ...   False       0.0   
2          NU              U                D1  ...   False       0.0   
3         NaN             R2                D1  ...   False       0.0   
4           N             R2                D1  ...   False       0.0   
..        ...            ...               ...  ...     ...       ...   
486        NC             R2                D2  ...   False       0.0   
487        NU             R1              D1.5  ...   False       0.0   
488         N             R2                D2  ...   False       0.0   
489        NC             R3                D3  ...   False       0.0   
490         N             R1                D2  ...   False       0.0   

    deb_hwy_d                                          avcomment  edcr_user  \
0         0.0  Wet slab released on SE aspect @ 5400 feet (16...         ed   
1         0.0  Wet debris ran 2/3 path in main path to 4700 f...         ed   
2         0.0  Terminus at canyon floor elevation.\r\n\r\nDeb...         ed   
3         0.0                                               D1.5         ed   
4         0.0                                    D1.5- 1/2 Path.         ed   
..        ...                                                ...        ...   
486       0.0  Slab not observed until March 17th during a sk...       Adam   
487       0.0                                                NaN       Adam   
488       0.0                          Obs on Mar. 24th from Hwy       Adam   
489       0.0  Obs on Mar. 24th from Hwy. Could not see runou...       Adam   
490       0.0                        Obs. on Mar. 28th from Hwy.       Adam   

              edcr_time       edcr_ip  eded_user            eded_time  eded_ip  
0   2016-03-17 11:50:56  10.173.132.2        NaN  0000-00-00 00:00:00      NaN  
1   2016-03-17 11:50:56  10.173.132.2        NaN  0000-00-00 00:00:00      NaN  
2   2016-03-17 11:50:56  10.173.132.2        NaN  0000-00-00 00:00:00      NaN  
3   2016-03-17 11:50:56  10.173.132.2        NaN  0000-00-00 00:00:00      NaN  
4   2016-03-17 11:50:56  10.173.132.2        NaN  0000-00-00 00:00:00      NaN  
..                  ...           ...        ...                  ...      ...  
486 2022-03-17 21:17:04    10.30.72.4        NaN  0000-00-00 00:00:00      NaN  
487 2022-03-23 21:32:42    10.30.72.4        NaN  0000-00-00 00:00:00      NaN  
488 2022-03-28 16:30:35    10.30.72.4        NaN  0000-00-00 00:00:00      NaN  
489 2022-03-28 16:34:43    10.30.72.4        NaN  0000-00-00 00:00:00      NaN  
490 2022-03-28 16:37:03    10.30.72.4        NaN  0000-00-00 00:00:00      NaN  

[491 rows x 39 columns]

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 491 entries, 0 to 490
Data columns (total 39 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   avrid              491 non-null    int64         
 1   avpathid           491 non-null    object        
 2   avobsdate          491 non-null    datetime64[ns]
 3   avobstime          491 non-null    object        
 4   avobserverid       491 non-null    object        
 5   avtype             485 non-null    object        
 6   avtriggertyp       455 non-null    float64       
 7   avtrigger          446 non-null    object        
 8   avrelativesize     474 non-null    object        
 9   avdestructivesize  441 non-null    object        
 10  avinterface        431 non-null    object        
 11  sz_vert            470 non-null    object        
 12  sz_horiz           453 non-null    object        
 13  sz_avgslope        491 non-null    int64         
 14  sz_elev_m 

In [10]:
df.columns

Index(['avrid', 'avpathid', 'avobsdate', 'avobstime', 'avobserverid', 'avtype',
       'avtriggertyp', 'avtrigger', 'avrelativesize', 'avdestructivesize',
       'avinterface', 'sz_vert', 'sz_horiz', 'sz_avgslope', 'sz_elev_m',
       'sz_aspect', 'term_elev_m', 'av_vert_m', 'crown_width_m', 'ch_max_cm',
       'ch_min_cm', 'ch_diff_cm', 'term_longpath', 'term_shortPath',
       'term_detail', 'term_moist', 'deb_rail', 'deb_rail_w', 'deb_rail_d',
       'deb_hwy', 'deb_hwy_w', 'deb_hwy_d', 'avcomment', 'edcr_user',
       'edcr_time', 'edcr_ip', 'eded_user', 'eded_time', 'eded_ip'],
      dtype='object')

useful:

##### I took out all of the user related columns that don't matter
<br />
['avrid', 'avpathid', 'avobsdate', 'avobstime', 'avobserverid', 'avtype',
       'avtriggertyp', 'avtrigger', 'avrelativesize', 'avdestructivesize',
       'avinterface', 'sz_vert', 'sz_horiz', 'sz_avgslope', 'sz_elev_m',
       'sz_aspect', 'term_elev_m', 'av_vert_m', 'crown_width_m', 'ch_max_cm',
       'ch_min_cm', 'ch_diff_cm', 'term_longpath', 'term_shortPath',
       'term_detail', 'term_moist', 'deb_rail', 'deb_rail_w', 'deb_rail_d',
       'deb_hwy', 'deb_hwy_w', 'deb_hwy_d']


In [25]:
# stands for data to edit

pd.set_option('display.max_columns', 500)


dte = df.loc[:, ['avrid', 'avpathid', 'avobsdate', 'avobstime', 'avobserverid', 'avtype',
       'avtriggertyp', 'avtrigger', 'avrelativesize', 'avdestructivesize',
       'avinterface', 'sz_vert', 'sz_horiz', 'sz_avgslope', 'sz_elev_m',
       'sz_aspect', 'term_elev_m', 'av_vert_m', 'crown_width_m', 'ch_max_cm',
       'ch_min_cm', 'ch_diff_cm', 'term_longpath', 'term_shortPath',
       'term_detail', 'term_moist', 'deb_rail', 'deb_rail_w', 'deb_rail_d',
       'deb_hwy', 'deb_hwy_w', 'deb_hwy_d']]

In [26]:
dte

avrid  avpathid  avobsdate avobstime avobserverid avtype  avtriggertyp  \
0        1    SHED10 2012-03-26  12:45:00     stein_01     WS           5.0   
1        2  PATH1163 2012-03-16  08:30:00     stein_01      U           5.0   
2        3     CURLY 2012-03-16  08:34:00     stein_01      U           5.0   
3        4    SHED10 2012-03-10  14:50:00     stein_01     WL           5.0   
4        5  INFINITY 2012-03-10  14:54:00     stein_01      L           5.0   
..     ...       ...        ...       ...          ...    ...           ...   
486    504   SHED07W 2022-03-01  05:04:00     clark_01     SS           5.0   
487    505   AP1163H 2022-03-23  13:30:00     clark_01     WL           5.0   
488    506    SHED10 2022-03-23  13:30:00     clark_01     WL           5.0   
489    507   MCAMSER 2022-03-23  18:00:00     clark_01     HS           5.0   
490    508  PATH1163 2022-03-27  14:35:00     clark_01     WL           5.0   

    avtrigger avrelativesize avdestructivesize avinterface sz_vert sz_horiz  \
0          NL             R1              D1.5           O       B        R   
1          NU              U                D2           U       U        U   
2          NU              U                D1           U       U        U   
3         NaN             R2                D1           I       M        R   
4           N             R2                D1           I       T        C   
..        ...            ...               ...         ...     ...      ...   
486        NC             R2                D2           I       T        L   
487        NU             R1              D1.5           O       T        U   
488         N             R2                D2           O       T        C   
489        NC             R3                D3           O       T        C   
490         N             R1                D2           O       T        L   

     sz_avgslope  sz_elev_m sz_aspect  term_elev_m  av_vert_m  crown_width_m  \
0              0          0       NaN            0          0              0   
1              0          0       NaN            0          0              0   
2              0          0       NaN            0          0              0   
3              0          0       NaN            0          0              0   
4              0          0       NaN            0          0              0   
..           ...        ...       ...          ...        ...            ...   
486           38       2015         E         1760        255              0   
487            0       1830        SW         1460        370              0   
488           40       2075        SE         1646        429              0   
489           40       2195        SE         1790        405            245   
490           40       2135        SE         1400        735              0   

     ch_max_cm  ch_min_cm  ch_diff_cm term_longpath term_shortPath  \
0            0          0           0            TK            NaN   
1            0          0           0            TK            NaN   
2            0          0           0            TR            NaN   
3            0          0           0            TK            NaN   
4            0          0           0            TK            NaN   
..         ...        ...         ...           ...            ...   
486          0          0           0            TK              -   
487          0          0           0            TR              -   
488          0          0           0            TK              -   
489        150         40          95            MR              -   
490          0          0           0            MR              -   

    term_detail term_moist  deb_rail  deb_rail_w  deb_rail_d  deb_hwy  \
0           NaN        NaN     False         0.0         0.0    False   
1           NaN        NaN     False         0.0         0.0    False   
2           NaN        NaN     False         0.0         0.0    False   
3           N

In [28]:
#dte.info()

 0   avrid              491 non-null    int64         
 1   avpathid           491 non-null    object        
 2   avobsdate          491 non-null    datetime64[ns]
 3   avobstime          491 non-null    object        
 4   avobserverid       491 non-null    object        
 5   avtype             485 non-null    object        
 6   avtriggertyp       455 non-null    float64       
 7   avtrigger          446 non-null    object        
 8   avrelativesize     474 non-null    object        
 9   avdestructivesize  441 non-null    object        
 10  avinterface        431 non-null    object        
 11  sz_vert            470 non-null    object        
 12  sz_horiz           453 non-null    object        
 13  sz_avgslope        491 non-null    int64         
 14  sz_elev_m          491 non-null    int64         
 15  sz_aspect          105 non-null    object        
 16  term_elev_m        491 non-null    int64         
 17  av_vert_m          491 non-null    int64         
 18  crown_width_m      491 non-null    int64         
 19  ch_max_cm          491 non-null    int64         
 20  ch_min_cm          491 non-null    int64         
 21  ch_diff_cm         491 non-null    int64         
 22  term_longpath      461 non-null    object        
 23  term_shortPath     326 non-null    object        
 24  term_detail        307 non-null    object        
 25  term_moist         378 non-null    object        
 26  deb_rail           491 non-null    bool          
 27  deb_rail_w         491 non-null    float64       
 28  deb_rail_d         491 non-null    float64       
 29  deb_hwy            491 non-null    bool          
 30  deb_hwy_w          491 non-null    float64       
 31  deb_hwy_d          491 non-null    float64       

**Do they have nulls?**

None: 'avrid', 'avpathid', 'avobsdate', 'avobstime', 'avobserverid' 

Some: 'avtype', 'avtriggertyp', 'avtrigger', 'avrelativesize', 'avdestructivesize', 'avinterface', 'sz_vert', 'sz_horiz', 
       
None: 'sz_avgslope', 'sz_elev_m',

Some: 'sz_aspect', 

None: 'term_elev_m', 'av_vert_m', 'crown_width_m', 'ch_max_cm', 'ch_min_cm', 'ch_diff_cm', 
       
Some: 'term_longpath', 'term_shortPath',
       'term_detail', 'term_moist', 
       
None: 'deb_rail', 'deb_rail_w', 'deb_rail_d', 'deb_hwy', 'deb_hwy_w', 'deb_hwy_d'

**Complete list:**

None: 
'avrid', 'avpathid', 'avobsdate', 'avobstime', 'avobserverid', 'sz_avgslope', 'sz_elev_m', 'term_elev_m', 'av_vert_m', 'crown_width_m', 'ch_max_cm', 'ch_min_cm', 'ch_diff_cm', 'deb_rail', 'deb_rail_w', 'deb_rail_d', 'deb_hwy', 'deb_hwy_w', 'deb_hwy_d'

Some: 'avtype', 'avtriggertyp', 'avtrigger', 'avrelativesize', 'avdestructivesize', 'avinterface', 'sz_vert', 'sz_horiz', 'sz_aspect',  'term_longpath', 'term_shortPath',
'term_detail', 'term_moist', 
       